### PR TITLE
Refactor Embeds to use custom components

### PR DESCRIPTION
**Pull request by Tirokk**
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-17T12:30:32Z" title="Tuesday, September 17th 2019, 2:30:32 pm +02:00">Sep 17, 2019</time>_
_Closed <time datetime="2019-10-14T12:09:27Z" title="Monday, October 14th 2019, 2:09:27 pm +02:00">Oct 14, 2019</time>_
---

### Todo

- [X] None

### CODE REVIEW

File: backend/src/schema/resolvers/embeds.js [hunk 1]
```diff
@@ -11,16 +11,17 @@ export default {
...
       'publisher',
-      'date',
-      'description',
       'url',
+      'date',
       'image',
       'audio',
       'video',
       'lang',
-      'html',
```
Comment (Tirokk) [inline backend/src/schema/resolvers/embeds.js:23]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-17T16:15:23Z" title="Tuesday, September 17th 2019, 6:15:23 pm +02:00">Sep 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -11,16 +11,17 @@ export default {
     ...undefinedToNullResolver([
       'type',
       'title',
+      'description',
       'author',
       'publisher',
-      'date',
-      'description',
       'url',
+      'date',
       'image',
       'audio',
       'video',
       'lang',
-      'html',
```


Hm, I think for the trusted oembed providers in `providers.json` we can return plain html to the client.

File: webapp/components/Embeds/VideoEmbed.vue [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    },
+  },
+  data() {
+    return {
+      renderEmbed: false,
+    }
+  },
+  computed: {
+    embedUrl() {
+      if (this.embedData.publisher === 'YouTube') {
```
Comment (Tirokk) [inline webapp/components/Embeds/VideoEmbed.vue:50]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-17T16:25:21Z" title="Tuesday, September 17th 2019, 6:25:21 pm +02:00">Sep 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,110 @@
+<template>
+  <div class="video-embed">
+    <div class="video-embed-wrapper" v-if="embedUrl">
+      <div v-if="!renderEmbed" class="video-embed-preview" @click="renderEmbed = true">
+        <img :src="embedData.image" />
+        <div class="player-icon">
+          <ds-icon name="youtube-play" size="xxx-large" />
+        </div>
+      </div>
+      <iframe
+        v-else
+        :src="url"
+        frameborder="0"
+        webkitallowfullscreen
+        mozallowfullscreen
+        allowfullscreen
+      ></iframe>
+    </div>
+    <default-embed v-else :previewImage="embedData.image" :url="url" />
+  </div>
+</template>
+
+<script>
+import DefaultEmbed from './DefaultEmbed.vue'
+
+export default {
+  name: 'video-embed',
+  components: {
+    DefaultEmbed,
+  },
+  props: {
+    url: {
+      type: String,
+      default: '',
+    },
+    embedData: {
+      type: Object,
+      default() {
+        return {}
+      },
+    },
+  },
+  data() {
+    return {
+      renderEmbed: false,
+    }
+  },
+  computed: {
+    embedUrl() {
+      if (this.embedData.publisher === 'YouTube') {
```


No, I don't want to see this kind of complexity. I would prefer to stick to the current solution for more complex embeds. 

 For security, have a look if we can trust the oembed provider. For data privacy, wrap the embed with an additional click (are you OK to load stuff from 3rd parties?). For mobile responsiveness, try out if the iframe or whatever breaks the layout.

File: webapp/components/Editor/nodes/Embed.spec.js [hunk 3]
```diff
@@ -9,23 +9,17 @@ describe('Embed.vue', () => {
   beforeEach(() => {
```
Comment (Tirokk) [inline webapp/components/Editor/nodes/Embed.spec.js:9]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-17T16:28:18Z" title="Tuesday, September 17th 2019, 6:28:18 pm +02:00">Sep 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -9,23 +9,17 @@ describe('Embed.vue', () => {
   beforeEach(() => {
```


I love that you created test cases for the embeds :heart:

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-14T12:09:27Z" title="Monday, October 14th 2019, 2:09:27 pm +02:00">Oct 14, 2019</time>_

---


closing this in favor of https://github.com/Human-Connection/Human-Connection/pull/1686
we could maybe still refactor in the future to have separate components based on what is returned from `oEmbed`...

**Review by Mogge — CHANGES_REQUESTED** [review]
Here's the result of testing the following youtube link: https://youtu.be/CFJTnWppNVI

Screenshot - 2019-09-17T183154 415


Screenshot - 2019-09-17T183232 129


This is the reason why I don't like complexity or the attempt to outsmart the embedded code from 3rd parties like Youtube. Our code will soon break once they change their API.

I suggest to stick with what we have right now for complex embeds like videos etc. Simply check the oembed `providers.json`!  I'm also OK to remove css class on the client if the iframes mess up our layout.

**Pull request closed without merging**